packages/volto/razzle.config.js

```diff
@@ -156,6 +158,51 @@ const defaultModify = ({
         placeholders: true,
       }),
     );
+
+    // This copies the publicPath files set in voltoConfigJS with the local `public`
+    // directory at build time
+    config.plugins.push(
+      new AfterBuildPlugin(() => {
+        const mergeDirectories = (sourceDir, targetDir) => {
+          const files = fs.readdirSync(sourceDir);
+          files.forEach((file) => {
+            const sourcePath = path.join(sourceDir, file);
+            const targetPath = path.join(targetDir, file);
+            fs.copyFileSync(sourcePath, targetPath);
+          });
+        };
+
+        registry.getAddonDependencies().forEach((addonDep) => {
+          // What comes from getAddonDependencies is in the form of `@package/addon:profile`
+          const addon = addonDep.split(':')[0];
+          // Check if the addon is available in the registry, just in case
+          if (registry.packages[addon]) {
+            const p = fs.realpathSync(
+              `${registry.packages[addon].modulePath}/../.`,
+            );
+            if (fs.existsSync(path.join(p, 'public'))) {
+              if (!dev) {
+                mergeDirectories(path.join(p, 'public'), paths.appBuildPublic);
+              }
+              if (
+                dev &&
+                !registry.isVoltoProject &&
+                registry.addonNames.length > 0
+              ) {
+                const devPublicPath = `${projectRootPath}/../../../public`;
+                if (!fs.existsSync(devPublicPath)) {
+                  fs.mkdirSync(devPublicPath);
+                }
+                mergeDirectories(
+                  path.join(p, 'public'),
+                  `${projectRootPath}/../../../public`,
```

Comment:
This assumes a development structure, and I think I am ok with the idea, but I don't understand which structure it assumes you are using for developing the add-on.

Reply:
It assumes that:
- You are in dev (pnpm start)
- You are running Volto in a non-project, old fashioned structure. 
- You have add-ons, so it means that you are in the new setup. (Unlikely that you are using Volto as standalone with add-ons, in a setup that is not the new setup)

```
    dev &&
   !registry.isVoltoProject &&
   registry.addonNames.length > 0
```

So I think it's safe and would only trigger if you are using the new setup. This can be a drawback because the dev mode will only work in the new setups, however, the build will continue harvest the public folders and do the right thing.

Reply:
Ok, understood, makes sense.